data-prepper-plugins/aws-sqs-common/src/main/java/org/opensearch/dataprepper/plugins/aws/sqs/common/SqsService.java

```diff
@@ -0,0 +1,143 @@
+                .withMaxAttempts(Integer.MAX_VALUE);
+        this.bufferAccumulator = bufferAccumulator;
+        this.codec = codec;
+    }
+
+    SqsClient createSqsClient(final Region region, final AwsCredentialsProvider awsCredentialsProvider) {
+        LOG.info("Creating SQS client");
+        return SqsClient.builder()
+                .region(region)
+                .credentialsProvider(awsCredentialsProvider)
+                .overrideConfiguration(ClientOverrideConfiguration.builder()
+                        .retryPolicy(builder -> builder.numRetries(5).build())
+                        .build())
+                .build();
+    }
+
+    public ReceiveMessageRequest createReceiveMessageRequest(final SqsOptions sqsOptions) {
+        return ReceiveMessageRequest.builder()
+                .queueUrl(sqsOptions.getSqsUrl())
+                .maxNumberOfMessages(sqsOptions.getMaximumMessages())
+                .build();
+    }
+
+    public List<Message> getMessagesFromSqs(final SqsOptions sqsOptions) {
+        try {
+            final ReceiveMessageRequest receiveMessageRequest = createReceiveMessageRequest(sqsOptions);
+            final List<Message> messages = sqsClient.receiveMessage(receiveMessageRequest).messages();
+            if(!messages.isEmpty())
+                sqsMetrics.getSqsMessagesReceivedCounter().increment(messages.size());
+            failedAttemptCount = 0;
+            return messages;
+        } catch (final SqsException | StsException e) {
+            LOG.error("Error reading from SQS: {}. Retrying with exponential backoff.", e.getMessage());
+            applyBackoff();
+            return Collections.emptyList();
+        }
+    }
+
+    public void applyBackoff() {
+        final long delayMillis = backoff.nextDelayMillis(++failedAttemptCount);
+        if (delayMillis < 0) {
+            Thread.currentThread().interrupt();
+            throw new SqsRetriesExhaustedException("SQS retries exhausted. Make sure that SQS configuration is valid, SQS queue exists, and IAM role has required permissions.");
+        }
+        final Duration delayDuration = Duration.ofMillis(delayMillis);
+        LOG.info("Pausing SQS processing for {}.{} seconds due to an error in processing.",
+                delayDuration.getSeconds(), delayDuration.toMillisPart());
+        try {
+            Thread.sleep(delayMillis);
+        } catch (final InterruptedException e){
+            LOG.error("Thread is interrupted while polling SQS with retry.", e);
+        }
+    }
+
+    public Optional<String> parseMessage(final Message message,final AcknowledgementSet acknowledgementSet) throws IOException {
+        pushToBuffer(codec.parse(message.body()),acknowledgementSet);
+        return Optional.of(message.receiptHandle());
+    }
+
+    public boolean pushToBuffer(final Record<Event> eventRecord, final AcknowledgementSet acknowledgementSet) {
```

Comment:
I don't think this common code should push to the buffer. This is not the right way for this to work with S3.

Reply:
Thanks @dlvenable for your review, Sure i will make the changes accordingly.